README.md

```diff
@@ -1,54 +1,58 @@
+[Status Network](https://status.network) is the **first natively gasless Ethereum L2**, optimized for social apps and games, featuring sustainable public funding for developers through native yield and DEX fees. Built on the Linea zkEVM stack, it provides high-performance, gas-free transactions while ensuring economic sustainability through a novel funding model and spam prevention technology.
 
-## What is Linea?
+## Gasless Transaction System
 
-[Linea](https://linea.build) is a developer-ready layer 2 network scaling Ethereum. It's secured with a zero-knowledge rollup, built on lattice-based cryptography, and powered by [Consensys](https://consensys.io).
+Status Network introduces **gasless transactions** through a **RLN technology** and **Karma reputation system**. This allows users to submit transactions without paying gas fees while maintaining network security and preventing spam.
 
+### How It Works
 
-## Get started
+**Rate Limiting Nullifier**: A cryptographic system that prevents spam by limiting transaction rates through nullifier-based proofs. Implementation can be found [here](https://github.com/vacp2p/zerokit).
 
-If you already have an understanding of the tech stack, use our [Get Started](docs/get-started.md) guide.
+**火 Karma System**: A reputation-based mechanism where users earn Karma soulbound tokens through positive network participation. Users will have different levels of daily gasless transaction quota depending on their Karma amount. Contract code implementation can be found [here](https://github.com/vacp2p/staking-reward-streamer).
 
-For developers looking to build services locally (e.g., the coordinator), see our detailed [Local Development Guide](docs/local-development-guide.md).
+**Premium Gas Bypass**: When users exceed their daily gasless transaction quota, they can still submit transactions by paying premium gas fees. The premium gas threshold is configurable and allows users to bypass rate limiting restrictions when needed.
 
-## Looking for the Linea code?
+### Architecture Components
 
-Linea's stack is made up of multiple repositories, these include:
+#### Besu Plugin Components
+- [**TxForwarder Transaction Pool Validator**](besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/txpoolvalidation/validators/RlnProverForwarderValidator.java): Forwards incoming transaction data to the RLN prover service to generate RLN proofs
+- [**Modified LineaEstimateGas RPC**](besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/rpc/methods/LineaEstimateGas.java): Dynamically provides zero gas or premium gas estimates based on real-time user karma and usage quotas
+- [**Karma Service Integration**](besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/txpoolvalidation/shared/KarmaServiceClient.java): Real-time user tier and quota checking via gRPC
```

Comment:
Karma service is part of the RLN Prover, its client is part of the sequencer. It is also not a besu plugin.

Reply:
[1a997a5](https://github.com/status-im/status-network-monorepo/pull/4/commits/1a997a5fc5d623671f86d88824d1a22791b2550f)
Included the rln prover repo under the TxForwarder description along with a mention of Karma.
Removed explaining KarmaServiceClient as component.